docs/javascript/v2/how--to-guides/record-and-live-stream/hls/hls-timed-metadata.mdx

```diff
@@ -42,74 +42,153 @@ start_date and end_date will be in UTC format.
-    const cue = metaTextTrack.cues[cueIndex];
+    const cue: TextTrackCue = cues[cueIndex];
     if (cue.fired) {
       cueIndex++;
       continue;
     }
-    const data = metadataPayloadParser(cue.value.data);
-    const programData = videoEl.getStartDate();
+    // here we are converting base64 to actual data.
+    const data: Record<string, any> = metadataPayloadParser(cue.value.data);
+    const programData = videoEl?.getStartDate();
     const startDate = data.start_date;
     const endDate = data.end_date;
     const startTime =
-      new Date(startDate) -
-      new Date(programData) -
-      videoEl.currentTime * 1000;
-    const duration = new Date(endDate) - new Date(startDate);
+      new Date(startDate).getTime() - new Date(programData).getTime() - (videoEl?.currentTime || 0) * 1000;
+    const duration = new Date(endDate).getTime() - new Date(startDate).getTime();
     setTimeout(() => {
       const toast = {
         title: `Payload from timed Metadata ${data.payload}`,
         duration: duration,
       };
       console.debug("Added toast ", JSON.stringify(toast));
-      ToastManager.addToast(toast);
     }, startTime);
     cue.fired = true;
     cueIndex++;
   }
 };
+const handleTimeUpdateListener = () => {
+  // extract timed metadata text track
+  const metaTextTrack: TextTrack | null = extractMetaTextTrack();
+  if (!metaTextTrack || !metaTextTrack.cues) {
+    return;
+  }
+  // fire cue for timed meta data extract
+  fireCues(metaTextTrack.cues);
+};
 if (Hls.isSupported()) {
   /**
    * initialize HLSController and add event listeners.
    */
-  hlsController = new HLSController(hlsUrl, videoRef);
-  hlsStats = new HlsStats(hlsController.getHlsJsInstance(), videoEl);
-
-  hlsController.on(HLS_STREAM_NO_LONGER_LIVE, handleNoLongerLive);
-  hlsController.on(HLS_TIMED_METADATA_LOADED, metadataLoadedHandler);
-
-  hlsController.on(Hls.Events.MANIFEST_LOADED, manifestLoadedHandler);
-  hlsController.on(Hls.Events.LEVEL_UPDATED, levelUpdatedHandler);
-  hlsController.on(Hls.Events.ERROR, handleHLSError);
+  hls = new Hls();
+  hls.loadSource(hlsUrl);
+  hls.attachMedia(videoEl);
+  hls.on(Hls.Events.MANIFEST_LOADED, manifestLoadedHandler);
+  hls.on(Hls.Events.LEVEL_UPDATED, levelUpdatedHandler);
+  hls.on(Hls.Events.FRAG_CHANGED, fragChangeHandler);
 } else if (videoEl.canPlayType("application/vnd.apple.mpegurl")) {
-    // Work where hls.js is not supported
-    videoEl.src = hlsUrl;
-    videoEl.addEventListener("timeupdate", handleTimeUpdateListener);
-    setIsMSENotSupported(true);
+  // Work where hls.js is not supported
+  videoEl.src = hlsUrl;
+  videoEl.addEventListener("timeupdate", handleTimeUpdateListener);
 }
-
 ```
 
->Note : Above implementation might change in future.
-
 ## Limitations
 
 - It’s hard to sync metadata with stream exactly due to the latency from FFmpeg creating the stream. To address this, we’ve added 2 seconds delay of the actual `START-DATE` attribute.
```

Comment:
on the last limitation, what if someone wants to send a json? can they use the base64 workaround?

Reply:
That limitation is no longer there. It'll be removed with today's backend release to preprod - possibly IN release on monday night.

some context: https://100ms-live.slack.com/archives/C0409JW6Y75/p1677049901459419

Reply:
Even the `The API now allows sending only three metadata objects at once` limitation needs to be removed after sdk changes to accept only one payload at a time are made.
cc: @saikatmitra91 @amar-1995